### PR TITLE
makefile: Added help target that prints available targets

### PR DESCRIPTION
**Pull request by sjalander**
Started adding help make targets such that a user easily can see what targets that exist.

### CODE REVIEW

**Comment by sjalander** [comment]
Also, fixed a problem with compilation flags when trying to compile jive from jlm

**Comment by phate** [comment]
Several minor issues:

1. The clean command does not remove tests/test-runner. I think you need to replace rm -rf $(TESTPRROGS) in tests/Makefile.sub with rm -rf tests/$(TESTPROGS)

2. You assign test-runner to the variable TESTPROGS in tests/Makefile.sub, but then you continue to use test-runner instead of TESTPROGS throughout tests/Makefile.sub.

3. I would like it if you could be more precise with the help message of clean. Something like: Removes all build files, libraries and executables. The problem I currently have is that jive-clean removes the build files for jive and the jive library, but ALSO the build files of the tests. The only thing it does not seem to remove is tests/test-runner. Maybe you should remove the jive-clean from the help text completely and just have clean, and clean would then remove everything (irregardless of whether it was built in the first place). That is maybe the easiest.

4. The help message of the all target. Maybe you can reword it to: Compiles jive library and runs all unit tests

**Comment by sjalander** [comment]
'jive-clean' now deletes tests/test-runner and removed 'jive-test-clean'.

The tests/Makefile.sub has not been converted to have "generic" paths so when including the jive/Makefile.sub from somewhere else it is not possible to run the checks, i.e., it's not possible to run jive checks from the jlm root directory. Jive checks are only possible from the jive root directory.

'all' and 'clean' targets are "aliases" that exists in all 'root' directories. So both jive and jlm will have 'all' and 'clean' but 'all' in jive means 'clean-jive' while 'all' in jlm means 'jive-clean jlm-clean'.

Removed TESTPROGS variable as it didn't serve a purpose.
